drivers: sbs_gauge: Add support for AtRate properties

madhurimaparuchuri opened the pull request:
BatteryMode(w), AtRate(r/w), AtRateTimeToFull(r), AtRateTimeToEmpty(r) and AtRateOK(r)

madhurimaparuchuri committed:
drivers: sbs_gauge: Add support for AtRate properties

BatteryMode(w), AtRate(r/w), AtRateTimeToFull(r), AtRateTimeToEmpty(r)
and AtRateOK(r)

Signed-off-by: Madhurima Paruchuri <mparuchuri@google.com>

aaronemassey commented on include/zephyr/drivers/fuel_gauge.h:
```diff
@@ -37,45 +37,53 @@ extern "C" {
 
 /** Battery current (uA); negative=discharging */
-#define FUEL_GAUGE_CURRENT		FUEL_GAUGE_AVG_CURRENT + 1
+#define FUEL_GAUGE_CURRENT		    FUEL_GAUGE_AVG_CURRENT + 1
```
Was this from a `clang-format`? If so, please have that as a separate commit.

madhurimaparuchuri replied:
Separated in new revision

aaronemassey commented on drivers/fuel_gauge/sbs_gauge/sbs_gauge.c:
```diff
@@ -95,9 +95,9 @@ static int sbs_gauge_get_prop(const struct device *dev, struct fuel_gauge_get_pr
 		rc = sbs_cmd_reg_read(dev, SBS_GAUGE_CMD_VOLTAGE, &val);
 		prop->value.voltage = val * 1000;
 		break;
-	case FUEL_GAUGE_MODE:
```
Let's do renaming in a separate PR. I'm happy to care of that too since

madhurimaparuchuri replied:
I renamed only at rate properties in a separate commit in this PR as it needs to be done anyways

madhurimaparuchuri committed:
drivers: sbs_gauge: Rename atRate properties to include 'SBS'

Add the term 'SBS' in atRate properties and Battery mode property name,
as they are specific to SBS

Signed-off-by: Madhurima Paruchuri <mparuchuri@google.com>

madhurimaparuchuri committed:
drivers: sbs_gauge: Reformat the changes

Reformat the code changes using clang

Signed-off-by: Madhurima Paruchuri <mparuchuri@google.com>

aaronemassey reviewed (approved):
@teburd Can you also review?

teburd reviewed (approved):
Looks fine to me, first two commits could possibly be squashed into a Add SBS specific attributes for AtRate, etc. Not a blocker.

@rgundi is likely interested in this one
